main.go

```diff
@@ -302,6 +302,17 @@ func setupReconcilers(ctx context.Context, mgr ctrl.Manager) {
 		os.Exit(1)
 	}
 
+	var runtimeRegistry runtimeregistry.ExtensionRegistry
+	var runtimeClient runtimeclient.Client
+	if feature.Gates.Enabled(feature.RuntimeSDK) {
+		// This is the creation of the single RuntimeExtension registry for the controller.
+		runtimeRegistry = runtimeregistry.New()
+		runtimeClient = runtimeclient.New(runtimeclient.Options{
+			Catalog:  catalog,
+			Registry: runtimeRegistry,
+		})
+	}
```

Comment:
```suggestion
	var runtimeClient runtimeclient.Client
	if feature.Gates.Enabled(feature.RuntimeSDK) {
		runtimeClient = runtimeclient.New(runtimeclient.Options{
			Catalog:  catalog,
			Registry: runtimeregistry.New(),
		})
	}
```
This enforces that the runtimeRegistry cannot be used directly (which is slightly safer and less code)